lib/scsi_defs.h

```diff
@@ -80,5 +88,60 @@ namespace scsi_defs {
     uint8_t vendorSpecific : 2;
   } ControlByte;
 
+  // SCSI Reference Manual Table 148 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  enum class SelectReport : uint8_t {
+    kRestrictedMethods = 0x0,
+    kWellKnown = 0x1,
+    kAllLogical = 0x2
+  };
+
+  // SCSI Reference Manual Table 147 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct ReportLunsCommand {
+    OptCode optCode = OptCode::kReportLuns;
+    uint8_t reserved1[1];
+    SelectReport selectReport;
+    uint8_t reserved2[3];
+    uint32_t allocLength;
+    uint8_t reserved3[1];
+    ControlByte controlByte;
+  } ABSL_ATTRIBUTE_PACKED;
+
+  // SCSI Reference Manual Table 149 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct ReportLunsData {
+    std::vector<LunAddress> lunList;
```

Comment:
Not sure how to allow for an arbitrarily-sized LUN list while also keeping the code lean. Any thoughts would be appreciated.

Reply:
I have the same question. Would love to hear what others think

Reply:
I'd favor defining plain helper functions to parse and/or serialize (as needed) the sequence to/from bytes, rather than writing a type that, unlike all the others, isn't POD and can't be bitcast.

Reply:
The standard approach is to have a struct for the header and a struct for the list elements, and then to have the code know how to glue them together.